File: dev/test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/reference_key_values_equal.go
// Check that the two references, `that` and `other`, are equal by
// comparing their [aastypes.Reference.Keys] by [aastypes.Key.Value]'s.
func ReferenceKeyValuesEqual(
	that aastypes.IReference,
	other aastypes.IReference) bool {
	thatKeys := that.Keys()
	otherKeys := other.Keys()

	if len(thatKeys) != len(otherKeys) {
		return false
	}

	for i, thatKey := range thatKeys {
		otherKey := otherKeys[i]

		if thatKey.Value() != otherKey.Value() {
			return false
		}
	}

	return true
}

File: dev/test_data/test_main/typescript/aas_core_meta.v3/input/snippets/Verification/reference_key_values_equal.ts
/**
 * Check that the two references, `that` and `other`,
 * are equal by comparing their {@link types.Reference.keys}
 * by {@link types.Key.value}'s.
 *
 * @param that - reference to be compared
 * @param other - to be compared against
 * @returns `true` if the key values are are equal
 */
export function referenceKeyValuesEqual(
  that: AasTypes.Reference,
  other: AasTypes.Reference
): boolean {
  if (that.keys.length != other.keys.length) {
    return false;
  }

  for (let i = 0; i < that.keys.length; i++) {
    if (that.keys[i].value !== other.keys[i].value) {
      return false;
    }
  }

  return true;
}
